modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -261,7 +298,7 @@ func (h *HTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 func (h *HTTPTransport) SetScheme(req *http.Request) {
 	if req.URL.Scheme == "" {
 		req.URL.Scheme = "http"
-		if h.TLS != nil {
+		if h.TLS != nil && req.URL.Port() != "80" {
```

Comment:
So, I've read https://github.com/caddyserver/caddy/issues/4834#issuecomment-1153694521 but I'm still having a hard time understanding why this port 80 check is necessary. If the transport is configured for HTTPS, it should probably use HTTPS, right? (This is a reverse proxy client after all, not the server or a web browser.)

Just wondering, if we leave it as-is, does this only break dynamic upstreams that try to use HTTPS on port 80? (If so, that's probably OK, since that's in violation of multiple specs I'm pretty sure.)

Reply:
I don't think it makes sense at all for SRV to be giving you both HTTP and HTTPS endpoints. That's a system design issue. It should only do one or the other, not both.

Reply:
:thinking: Are you sure you aren't looking for a forward proxy @kresike ?

We could probably discuss the matter of port 80 in another issue, but I think the solution for a dynamic ServerName value here is looking good, if we want to start with that.

Reply:
@kresike Ah, I see. That's a very interesting use case indeed. Thanks for letting me know. (So often I have no idea how people are using this software and it kinda leaves me in the dark. So I really appreciate knowing what the specific needs are!)